src/perform_step/low_storage_rk_perform_step.jl

```diff
@@ -55,17 +55,22 @@ end
 
   # other stages
   for i in eachindex(A2end)
-    f(k, u, p, t+c2end[i]*dt)
+    @. tmp = A2end[i]*tmp
+    f(WilliamsonWrapper{typeof(tmp),typeof(dt)}(tmp,dt), u, p, t+c2end[i]*dt)
     integrator.destats.nf += 1
-    @. tmp = A2end[i]*tmp + dt*k
     @. u   = u + B2end[i]*tmp
   end
 
   f(k, u, p, t+dt)
   integrator.destats.nf += 1
 end
 
+struct WilliamsonWrapper{kType, dtType}
+  kref::kType
+  dt::dtType
+end
 
+Base.setindex!(a::WilliamsonWrapper{kType, dtType}, b::bType, c::cType) where {kType, dtType, bType, cType} = (a.kref[c] += a.dt * b) 
```

Comment:
How does this interact with linear algebra?

Reply:
All the linear algebra in RHS is untouched. Just replaces `du = ....` statements to `du += ...`. So the RHS is all the same.